Source/Xamarin/Prism.Forms/Services/DeviceService.cs

```diff
@@ -1,80 +1,130 @@
+		/// <summary>
+		/// Gets the kind of device that Xamarin.Forms is currently working on.
+		/// </summary>
+		public TargetIdiom Idiom
+		{
+			get { return Device.Idiom; }
+		}
 
-        /// <summary>
-        /// Invokes an action on the device main UI thread.
-        /// </summary>
-        /// <param name="action">The Action to invoke</param>
-        public void BeginInvokeOnMainThread(Action action)
-        {
-            Device.BeginInvokeOnMainThread(action);
-        }
+		/// <summary>
+		/// Gets the Platform (OS) that Xamarin.Forms is working on.
+		/// </summary>
+		[Obsolete("Platform is obsolete. Use RuntimePlatform instead.")]
+		public TargetPlatform Platform
+		{
+			get { return Device.OS; }
+		}
 
-        /// <summary>
-        /// Executes different actions depending on which Platform (OS) that Xamarin.Forms is working.
-        /// </summary>
-        /// <param name="iOS">Action to execute when running on iOS</param>
-        /// <param name="android">Action to execute when running on Android</param>
-        /// <param name="winPhone">Action to execute when running on WinPhone</param>
-        /// <param name="defaultAction">Action to execute if no Action was provided for the current Platform (OS)</param>
-        public void OnPlatform(Action iOS = null, Action android = null, Action winPhone = null, Action defaultAction = null)
-        {
-            Device.OnPlatform(iOS, android, winPhone, defaultAction);
-        }
+		/// <summary>
+		/// Gets the Platform (OS) that the application is running on.  This is the native Device.RunTimePlatform property.
+		/// </summary>
+		public string DeviceRuntimePlatform
+		{
+			get { return Device.RuntimePlatform; }
+		}
 
-        /// <summary>
-        /// Returns different values depending on the Platform (OS) that Xamarin.Forms is working.
-        /// </summary>
-        /// <typeparam name="T">Type of value to be returned</typeparam>
-        /// <param name="iOS">The value for iOS</param>
-        /// <param name="android">The value for Android</param>
-        /// <param name="winPhone">The value for WinPhone</param>
-        /// <returns>The value for the current Platform (OS)</returns>
-        public T OnPlatform<T>(T iOS, T android, T winPhone)
-        {
-            return Device.OnPlatform<T>(iOS, android, winPhone);
-        }
+		/// <summary>
+		/// Gets the Platform (OS) that the application is running on. The result is an enum of type RuntimePlatform.
+		/// </summary>
+		public RuntimePlatform RuntimePlatform
+		{
+			get
+			{
+				switch (Device.RuntimePlatform)
+				{
+					case Android:
+						return RuntimePlatform.Android;
+					case iOS:
+						return RuntimePlatform.iOS;
+					case macOS:
+						return RuntimePlatform.macOS;
+					case Tizen:
+						return RuntimePlatform.Tizen;
+					case UWP:
+						return RuntimePlatform.UWP;
+					case WinPhone:
+						return RuntimePlatform.WinPhone;
+					case WinRT:
+						return RuntimePlatform.WinRT;
```

Comment:
Based on this implementation it will only work with iOS and Android with XF 2.3.4. WinPhone and Windows are Runtime Platforms in 2.3.4, and Windows was removed in 2.3.5 and replaced with UWP and WinRT